### PR TITLE
[mongodb] Plan2: Match multiple database and collection names using a regular expression in MongoDB.

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
Match multiple database and collection names using a regular expression in MongoDB.
Support regex filter by databaseList and collectionList.

### CODE REVIEW

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/internal/MongoDBConnectorSourceTask.java [hunk 1]
```diff
@@ -226,4 +256,103 @@ private boolean isCopying() {
             throw new IllegalStateException("Cannot access isCopying field of SourceTask", e);
         }
     }
+
+    private void init(Map<String, String> props) {
```
Comment (leonardBang) [inline flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/internal/MongoDBConnectorSourceTask.java:260]:
```suggestion
    private void initCapturedCollections(MongoClient mongoClient, Map<String, String> props) {
```
and then we can put this method to `CollectionDiscoveryUtils`

File: flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+        List<String> actual = TestValuesTableFactory.getResults("mongodb_sink");
+        assertThat(actual, containsInAnyOrder(expected));
+
+        result.getJobClient().get().cancel().get();
+    }
+
+    /** match single database and multiple collections: collection = ^db0\.coll_b\d?$ . */
+    @Test
+    public void testMatchSingleDatabaseAndMultipleCollections() throws Exception {
```
Comment (leonardBang) [inline flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java:155]:
```suggestion
    public void testMatchSingleQualifiedCollectionPattern() throws Exception {
```

File: flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+        List<String> actual = TestValuesTableFactory.getResults("mongodb_sink");
+        assertThat(actual, containsInAnyOrder(expected));
+
+        result.getJobClient().get().cancel().get();
+    }
+
+    /** match single database and multiple collections: database = db0 collection = .*coll_b\d? . */
+    @Test
+    public void testMatchSingleDatabaseAndMultipleCollections1() throws Exception {
```
Comment (leonardBang) [inline flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java:194]:
```suggestion
    public void testMatchSingleDatabaseWithCollectionPattern() throws Exception {
```

File: docs/content/connectors/mongodb-cdc.md [hunk 4]
```diff
@@ -316,6 +318,24 @@ The MongoDB CDC connector is a Flink Source connector which will read database s
...
+'copy.existing.namespace.regex' = '^(db0\\.coll0|db1\\.coll1)$'
+```
+
+The config option `namespace.regex` is a regular expression that matches the namespaces from which to watch for changes.<br>
+
+In the following example, matches namespaces `db0.coll0` and `db1.coll1` to watch.
+
+```
+'namespace.regex' = '^(db0\\.coll0|db1\\.coll1)$'
+```
```
Comment (leonardBang) [inline docs/content/connectors/mongodb-cdc.md:337]: Do we still need this document?

Comment (Jiabao-Sun) [reply docs/content/connectors/mongodb-cdc.md]: Thanks, I missed it here.

File: flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java (hunk 3 above)
Comment (Jiabao-Sun) [reply flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java]: Fixed

File: flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java (hunk 2 above)
Comment (Jiabao-Sun) [reply flink-connector-mongodb-cdc/src/test/java/com/ververica/cdc/connectors/mongodb/table/MongoDBRegexFilterITCase.java]: Fixed

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/internal/MongoDBConnectorSourceTask.java (hunk 1 above)
Comment (Jiabao-Sun) [reply flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/internal/MongoDBConnectorSourceTask.java]: Fixed

**Review by leonardBang — APPROVED** [review]
Thanks @Jiabao-Sun for the update, LGTM. 
Please feel free to merge once the CI passed.

**Comment by Jiabao-Sun** [comment]
Thanks @leonardBang , @bobby-richard.
Merged into master.

**Pull request merged**